master/cdev.c

```diff
@@ -63,13 +63,14 @@ static int eccdev_mmap(struct file *, struct vm_area_struct *);
 #define VM_FAULT_CHANGE_VERSION KERNEL_VERSION(4, 10, 0)
 
 #if LINUX_VERSION_CODE >= PAGE_FAULT_VERSION
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(4,11,0))
-static int eccdev_vma_fault(struct vm_area_struct *, struct vm_fault *);
+#if LINUX_VERSION_CODE >= VM_FAULT_CHANGE_VERSION
```

Comment:
I think here is a possible issue, the old code checked for kernel version < 4.11.0 but the new code is >= 4.10.

Reply:
This is actually intended as it is also a fix

In kernel 4.9, variable is called virtual_address:
https://elixir.bootlin.com/linux/v4.9/source/include/linux/mm.h#L298

In 4.10, it is called address:
https://elixir.bootlin.com/linux/v4.10/source/include/linux/mm.h#L299

Reply:
I see, that was a recent try to fix. Thx for clearing.